Clientary actions & sources

feyzullah opened the pull request:
### Actions
* Create Client
* Create Expense
* Create Estimate
* Create Invoice
* Create Task
### Sources
* New Contact Created
* New Expense Created
* New Payment Created
* New Project Created
* New Staff Created
* New Task Created
* New Invoice Created
* New Client Created
* New Estimate Created

lcaresia commented on components/clientary/app/clientary.app.ts:
```diff
@@ -1,13 +1,114 @@
+      label: "Project ID",
+      description: "Project ID",
+      async options({ page }) {
+        page++;
+        return await utils.getAsyncOptions({
+          resourceFn: this.getRequestMethod("getProjects"),
+          page,
```
```suggestion
        return await utils.getAsyncOptions({
          resourceFn: this.getRequestMethod("getProjects"),
          page: page + 1,
```

lcaresia commented on components/clientary/app/clientary.app.ts:
```diff
@@ -1,13 +1,114 @@
+      label: "Invoice ID",
+      description: "Invoice ID",
+      async options({ page }) {
+        page++;
+        return await utils.getAsyncOptions({
+          resourceFn: this.getRequestMethod("getInvoices"),
+          page,
```
```suggestion
        return await utils.getAsyncOptions({
          resourceFn: this.getRequestMethod("getInvoices"),
          page: page + 1,
```

lcaresia reviewed (changes requested):
Just some fixes!

lcaresia reviewed (approved):
LGTM!

DilanAthukorala commented:
@feyzullah Please check the following improvements,

1. Create Estimate Action -`Estimate Item Attributes` should be a mandatory prop
![image](https://user-images.githubusercontent.com/102980146/209351876-5372370c-7b05-466a-857d-2acb094d150f.png)


2. Create Expense Action - `client_id` and `invoice_id` is not properly passed to Clientary
![image](https://user-images.githubusercontent.com/102980146/209353186-04713abf-6b1f-4608-b0e8-f667b9421e77.png)


3. Create Invoice Action - `Estimate Item Attributes` should be a mandatory prop
![image](https://user-images.githubusercontent.com/102980146/209353597-d4db763a-562c-43d8-9e50-1258cf791336.png)


4. Create Task Action - `client_id`  is not properly passed to Clientary
![image](https://user-images.githubusercontent.com/102980146/209354252-79bd97d3-4e06-4618-bdd6-d05b4b5e652b.png)

feyzullah commented:
> @feyzullah Please check the following improvements,
> 
> 1. Create Estimate Action -`Estimate Item Attributes` should be a mandatory prop
>    ![image](https://user-images.githubusercontent.com/102980146/209351876-5372370c-7b05-466a-857d-2acb094d150f.png)
> 2. Create Expense Action - `client_id` and `invoice_id` is not properly passed to Clientary
>    ![image](https://user-images.githubusercontent.com/102980146/209353186-04713abf-6b1f-4608-b0e8-f667b9421e77.png)
> 3. Create Invoice Action - `Estimate Item Attributes` should be a mandatory prop
>    ![image](https://user-images.githubusercontent.com/102980146/209353597-d4db763a-562c-43d8-9e50-1258cf791336.png)
> 4. Create Task Action - `client_id`  is not properly passed to Clientary
>    ![image](https://user-images.githubusercontent.com/102980146/209354252-79bd97d3-4e06-4618-bdd6-d05b4b5e652b.png)

Hi @DilanAthukorala I have fixed 1st and 3rd ones. They are still optional but you won't see that error if they are not given.
For 2nd and 4th ones, if `client_id` is given with the `project_id`, the project must have the related client in clients on Clientary.
In 2nd one, I was not able to pass `invoice_id` to Clientary under any condition, so I removed that prop from the action. Can you please check once more?

DilanAthukorala commented:
Verification completed, ready for release!

feyzullah commented:
/approve

alysonturing reviewed (approved):
LGTM!
